python/social_bot/teacher_tasks.py

```diff
@@ -205,3 +208,462 @@ def set_goal_name(self, goal_name):
         logging.debug('Setting Goal to %s', goal_name)
         self._goal_name = goal_name
 
+
+@gin.configurable
+class GroceryGroundGoalTask(GoalTask):
```

Comment:
Not necessary to address this in this PR, but this task doesn't contain any reference to the GroceryGround env.  Does it make sense to call this something like MultiGoalWithDistractionTask?

Reply:
Good point. I was considering decouple the tasks with specific env. Goal task and kicking-ball task are supposed to more general. I was also considering rename GroceryGround to PlayGround and remove it's dependency on specific task. So that we can assemble different tasks and different morphology of teachers in the playground through gin. 
Too many changes in one pull request are hard to review. Let's open another PR for these.